### DISCUSSION TITLE
Error 504

### DISCUSSION THREAD

**Discussion by DavidMoa26** [vercel/community#4788] [Help] [answered]
### Summary

Why do all routes in the app get a 504 error?

This Serverless Function has timed out.
504: GATEWAY_TIMEOUT
Code: FUNCTION_INVOCATION_TIMEOUT

The error is only in production

### Steps to Reproduce

 const getStartedHandler = () => {

    if (status === 'authenticated') {
      router.push('/dashboard')
    } else {
      signIn('google', { callbackUrl: '/dashboard' })
    }
  }

**Comment by amyegan** [comment]
There are some troubleshooting suggestions for the `504: FUNCTION_INVOCATION_TIMEOUT` error at https://vercel.com/docs/errors/application-error-codes#504:-function_invocation_timeout

Do you have an example that will let us see the problem in more detail?

**Comment by DavidMoa26** [comment]
What details can I give you to describe the problem?
This is a next js application (app router), the application has 5 pages and
a landing page, when I am in development mode everything works as usual,
there are no problems, errors or special warnings, after the deployment it
is not possible to go to pages except for the landing page, the error I get
is:

This Serverless Function has timed out.

Your connection is working correctly.

Vercel is working correctly.

504: GATEWAY_TIMEOUT
Code: FUNCTION_INVOCATION_TIMEOUT
ID: fra1::tpjs6-1699625056567-a2d550b17d26

In my opinion, the problem is not in the code itself, but in Vercel's
settings, maybe hubby plan is not suitable, I am looking for an unequivocal
answer

‫בתאריך יום ה׳, 9 בנוב׳ 2023 ב-19:15 מאת ‪Amy‬‏ ***@***.***
‬‏>:‬

**Reply by amyegan** [reply]
I'm sorry @DavidMoa26. I can't offer an unequivocal answer without further investigation into the specifics of your situation. If you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), then I can use it investigate further. 

[Max duration](https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration) for Hobby plan serverless functions is 10 seconds. If the function takes longer than 10 seconds to return, that could be the cause of the error. Missing or invalid environment variables are another common reason for timeout errors since the service cannot connect. 

There are other possible causes, and I do recommend the troubleshooting tips I linked before. They cover some more common causes of this error. 

https://vercel.com/docs/errors/application-error-codes#504:-function_invocation_timeout

**Comment by DavidMoa26** [comment, marked as answer]
I forgot to register the vercel domain in the next config file, sorry for
the trouble

**Reply by amyegan** [reply]
No trouble at all. I'm glad you got it working! 😄

**Reply by levipadre** [reply]
What did you register @DavidMoa26 and how, what Vercel domain? I may have the same issue. Hobby plan and all pages give me 504
